src/main/java/com/marklogic/kafka/connect/source/SerializedConstraintColumnHandler.java

```diff
@@ -0,0 +1,92 @@
+package com.marklogic.kafka.connect.source;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.datamovement.RowBatcher;
+import com.marklogic.client.ext.helper.LoggingObject;
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.client.io.StringHandle;
+import com.marklogic.client.row.RawPlanDefinition;
+import com.marklogic.client.row.RowManager;
+
+import java.util.Map;
+
+public class SerializedConstraintColumnHandler extends LoggingObject implements ConstraintColumnHandler {
+    private final ObjectMapper mapper = new ObjectMapper();
+    private final DatabaseClient databaseClient;
+
+    SerializedConstraintColumnHandler(DatabaseClient databaseClient) {
+        this.databaseClient = databaseClient;
+    }
+
+    @Override
+    public String injectConstraintIntoQuery(Map<String, Object> parsedConfig, String constraintColumn, String previousMaxConstraintColumnValue) {
+        String originalQuery = (String) parsedConfig.get(MarkLogicSourceConfig.SERIALIZED_QUERY);
+        String constrainedQuery = originalQuery;
+        if (previousMaxConstraintColumnValue != null) {
+            String constraintNodeString = "{\"ns\":\"op\", \"fn\":\"where\", \"args\":[{\"ns\":\"op\", \"fn\":\"gt\", \"args\":[{\"ns\":\"op\", \"fn\":\"col\", \"args\":[\"" + constraintColumn + "\"]}, \"" + previousMaxConstraintColumnValue + "\"]}]}";
+            try {
+                ObjectNode constraintNode = (ObjectNode) mapper.readTree(constraintNodeString);
+
+                ObjectNode queryRoot = (ObjectNode) mapper.readTree(originalQuery);
+                ArrayNode rootArgsArray = (ArrayNode) queryRoot.get("$optic").get("args");
+
+                rootArgsArray.insert(1, constraintNode);
+                constrainedQuery = mapper.writeValueAsString(queryRoot);
+            } catch (JsonProcessingException e) {
+                throw new RuntimeException(e);
```

Comment:
We usually want friendly contextual messages, so e.g. - `throw new RuntimeException("Unable to modify serialized query, cause: " + e.getMessage(), e)`